lib/tasks/indices.rake

```diff
@@ -243,4 +243,68 @@ You should run this task if the index schema has changed.
       end
     end
   end
+
+  desc "
+  Checking to see what's in mainstream that isn't in govuk
+  "
+  task :compare_govuk_and_mainstream_yo do
+    require 'gds_api/content_store'
+
+    client = Services.elasticsearch(hosts: SearchConfig.new.base_uri, timeout: 5.0)
+    publishing_api = Services.publishing_api
+    content_store = GdsApi::ContentStore.new(Plek.new.find('content-store'))
+
+    indices = {
+      mainstream: [],
+      govuk: []
+    }
+
+    indices.each do |index, ids|
+      ScrollEnumerator.new(
+        client: client,
+        search_body: {
+          fields: ["_id"],
+          query: { match_all: {} }
+        },
+        batch_size: 500,
+        index_names: index.to_s
+      ) { |hit| hit }.map do |hit|
+        id = hit['_id']
+        id.prepend('/') unless id[0] == '/'
+        ids << id
+      end
+    end
+
+    not_in_govuk = indices[:mainstream] - indices[:govuk]
+
+    redirects = 0
+    gones = 0
+
+    actual_missing = not_in_govuk
+
+    puts "*************************************************************"
+    puts "Checking #{actual_missing.count} pieces of non unpublishing type content"
+    # puts actual_missing
+    puts "*************************************************************"
+
+    actual_missing.reject! do |base_path|
+      begin
+        content_item = JSON.parse(`curl https://www.gov.uk/api/content#{base_path}`)
```

Comment:
Could you handle external content as well? It has a full URL as a document ID.

Reply:
Thanks, we have added in a check/counter.